### PR TITLE
populate parent upcoming, pending and completed quests

### PR DESCRIPTION
**Pull request by isabelleleeson**
hard coded data for now, will add requests to server soon
also will make it look better later
image
image
image

### CODE REVIEW

File: android/app/src/main/res/layout/quest_recycler_view_item.xml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    <TextView
+        android:id="@+id/quest_mandatory_text_view"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_below="@id/quest_image_view"
+        android:layout_marginTop="5dp"
+        android:textSize="17sp"
+        android:layout_marginBottom="10dp"
+        android:layout_marginStart="0dp"/>
```
Comment (jimzhaozhengwang) [inline android/app/src/main/res/layout/quest_recycler_view_item.xml]: is this needed?

File: android/app/src/main/res/layout/fragment_parent_quests_completed.xml [hunk 2]
```diff
@@ -1,12 +1,36 @@
...
+                android:layout_height="wrap_content"
+                android:layout_below="@id/completed_text_view"
+                android:layout_marginTop="10dp"
+                android:scrollbars="horizontal" />
+
+        </RelativeLayout>
+
+    </ScrollView>
+
+</RelativeLayout>
```
Comment (jimzhaozhengwang) [inline android/app/src/main/res/layout/fragment_parent_quests_completed.xml]: add new line

File: android/app/src/main/res/layout/fragment_parent_quests_pending.xml [hunk 3]
```diff
@@ -1,12 +1,36 @@
...
+                android:layout_marginTop="15dp"
+                android:textSize="25sp" />
+
+            <android.support.v7.widget.RecyclerView
+                android:id="@+id/pending_quests_recycler_view"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_below="@id/pending_text_view"
+                android:layout_marginTop="10dp"
+                android:scrollbars="horizontal" />
```
Comment (jimzhaozhengwang) [inline android/app/src/main/res/layout/fragment_parent_quests_pending.xml]: vertical?

File: android/app/src/main/res/layout/fragment_parent_quests_completed.xml [hunk 4]
```diff
@@ -1,12 +1,36 @@
...
+                android:layout_marginTop="15dp"
+                android:textSize="25sp" />
+
+            <android.support.v7.widget.RecyclerView
+                android:id="@+id/completed_quests_recycler_view"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_below="@id/completed_text_view"
+                android:layout_marginTop="10dp"
+                android:scrollbars="horizontal" />
```
Comment (jimzhaozhengwang) [inline android/app/src/main/res/layout/fragment_parent_quests_completed.xml]: vertical?

File: android/app/src/main/res/layout/fragment_parent_quests_upcoming.xml [hunk 5]
```diff
@@ -1,12 +1,54 @@
...
+                android:layout_marginTop="15dp"
+                android:textSize="25sp" />
+
+            <android.support.v7.widget.RecyclerView
+                android:id="@+id/overdue_quests_recycler_view"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_below="@id/overdue_text_view"
+                android:layout_marginTop="10dp"
+                android:scrollbars="horizontal" />
```
Comment (jimzhaozhengwang) [inline android/app/src/main/res/layout/fragment_parent_quests_upcoming.xml]: vertical?

File: android/app/src/main/res/layout/quest_recycler_view_item.xml [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    android:paddingVertical="15dp"
+    android:clickable="true"
+    android:focusable="true"
+    android:gravity="center">
+
+    <ImageView
+        android:id="@+id/quest_image_view"
+        android:layout_width="70dp"
+        android:layout_height="70dp"
+        android:layout_marginTop="0dp"
```
Comment (jimzhaozhengwang) [inline android/app/src/main/res/layout/quest_recycler_view_item.xml]: is this needed?

File: android/app/src/main/res/layout/quest_recycler_view_item.xml [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        android:layout_marginStart="0dp"/>
```
Comment (jimzhaozhengwang) [inline android/app/src/main/res/layout/quest_recycler_view_item.xml]: is this needed?

File: android/app/src/main/res/layout/quest_recycler_view_item.xml [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+    <TextView
+        android:id="@+id/quest_name_text_view"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="0dp"
+        android:layout_marginTop="00dp"
```
Comment (jimzhaozhengwang) [inline android/app/src/main/res/layout/quest_recycler_view_item.xml]: is this needed?

File: android/app/src/main/res/layout/quest_recycler_view_item.xml [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+        android:layout_width="70dp"
+        android:layout_height="70dp"
+        android:layout_marginTop="0dp"
+        android:layout_marginStart="0dp"/>
+
+    <TextView
+        android:id="@+id/quest_name_text_view"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="0dp"
```
Comment (jimzhaozhengwang) [inline android/app/src/main/res/layout/quest_recycler_view_item.xml]: is this needed?

File: android/app/src/main/res/layout/quest_recycler_view_item.xml [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+        android:gravity=""
+        android:layout="@id/quest_image_view"
+        android:textSize="20sp" />
+
+    <TextView
+        android:id="@+id/quest_owner_text_view"
+        android:layout_width="wrap_content"
+        android:layout_height="30dp"
+        android:layout_below="@id/quest_name_text_view"
+        android:layout_marginStart="0dp"
```
Comment (jimzhaozhengwang) [inline android/app/src/main/res/layout/quest_recycler_view_item.xml]: is this needed?

File: android/app/src/main/res/layout/quest_recycler_view_item.xml [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+        android:layout_marginTop="0dp"
```
Comment (jimzhaozhengwang) [inline android/app/src/main/res/layout/quest_recycler_view_item.xml]: is this needed?

File: android/app/src/main/res/layout/quest_recycler_view_item.xml [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+        android:gravity=""
+        android:textSize="17sp" />
+
+    <TextView
+        android:id="@+id/quest_exp_text_view"
+        android:layout_width="wrap_content"
+        android:layout_height="30dp"
+        android:layout_below="@id/quest_name_text_view"
+        android:layout_marginStart="30dp"
+        android:layout_marginTop="0dp"
```
Comment (jimzhaozhengwang) [inline android/app/src/main/res/layout/quest_recycler_view_item.xml]: is this needed?

File: android/app/src/main/res/layout/quest_recycler_view_item.xml (hunk 1 above)
Comment (isabelleleeson) [reply android/app/src/main/res/layout/quest_recycler_view_item.xml]: xml is real messy rn, I was gonna fix the layout later but I can clean up a bit

File: android/app/src/main/res/layout/fragment_parent_quests_pending.xml (hunk 3 above)
Comment (isabelleleeson) [reply android/app/src/main/res/layout/fragment_parent_quests_pending.xml]: fixed

**Comment by isabelleleeson** [comment]
cleaned up the xmls

File: android/app/src/main/java/com/chorestory/adapter/QuestRecyclerViewAdapter.java [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+        myViewHolder.questImageView.setImageResource(currentItem.getImageId());
+        // Quest name
+        myViewHolder.questNameTextView.setText(currentItem.getName());
+        // Quest owner
+        String owner = currentItem.getOwner() + "'s quest";
+        myViewHolder.questOwnerTextView.setText(owner);
+        // Quest mandatory/optional
+        String mandatory;
+        if (currentItem.getMandatory()) {
+            mandatory = "Mandatory";
```
[the remark this answers is not in the document]
Comment (jimzhaozhengwang) [reply android/app/src/main/java/com/chorestory/adapter/QuestRecyclerViewAdapter.java]: @isabelleleeson 
@keller00 would like you to put these in `strings.xml`

**Pull request merged**